bigtable/beam/batch-write-flow-control-example/src/main/java/bigtable/BatchWriteFlowControlExample.java

```diff
@@ -0,0 +1,237 @@
+
+    // SdkHarnessOptions loggingOptions = options.as(SdkHarnessOptions.class);
+    // loggingOptions.getSdkHarnessLogLevelOverrides()
+    //     .addOverrideForPackage(Package.getPackage("com.google.cloud.bigtable.data.v2.stub"),
+    //         LogLevel.DEBUG);
+
+    generateData(options);
+  }
+
+  static void generateData(BigtablePipelineOptions options) {
+
+    System.out.println("Generating " + options.getBigtableRows() + " rows, each "
+        + options.getBigtableColsPerRow() + " columns, " + options.getBigtableBytesPerCol()
+        + " bytes per column, " + options.getBigtableColsPerRow() * options.getBigtableBytesPerCol()
+        + " bytes per row.");
+
+    String generateLabel = String
+        .format("Generate %d rows for table %s", numRows, options.getBigtableTableId());
+    String mutationLabel = String
+        .format("Create mutations that write %d columns of total %d bytes to each row",
+            options.getBigtableColsPerRow(), options.getBigtableColsPerRow()
+                * options.getBigtableBytesPerCol());
+
+    Pipeline p = Pipeline.create(options);
+
+    PCollection<Long> numbers = p.apply(generateLabel, GenerateSequence.from(0).to(numRows));
+
+    if (options.getUseCloudBigtableIo()) {
+      System.out.println("Using CloudBigtableIO");
+      PCollection<org.apache.hadoop.hbase.client.Mutation> mutations = numbers.apply(mutationLabel,
+          ParDo.of(new CreateHbaseMutationFn(options.getBigtableColsPerRow(),
+              options.getBigtableBytesPerCol())));
+
+      mutations.apply(
+          String.format("Write data to table %s via CloudBigtableIO", options.getBigtableTableId()),
+          CloudBigtableIO.writeToTable(new CloudBigtableTableConfiguration.Builder()
+              .withProjectId(options.getProject())
+              .withInstanceId(options.getBigtableInstanceId())
+              .withTableId(options.getBigtableTableId())
+              .withConfiguration(BigtableOptionsFactory.BIGTABLE_ENABLE_BULK_MUTATION_FLOW_CONTROL,
+                  "true")
+              .withConfiguration(BigtableOptionsFactory.BIGTABLE_BULK_MAX_REQUEST_SIZE_BYTES,
+                  "1048576")
+              .build()));
+    } else {
+      System.out.println("Using BigtableIO");
+      PCollection<KV<ByteString, Iterable<Mutation>>>
+          mutations = numbers.apply(mutationLabel,
+          ParDo.of(new CreateMutationFn(options.getBigtableColsPerRow(),
+              options.getBigtableBytesPerCol())));
+
+      BigtableIO.Write write = BigtableIO.write()
+          .withProjectId(options.getProject())
+          .withInstanceId(options.getBigtableInstanceId())
+          .withTableId(options.getBigtableTableId())
+          .withFlowControl(true);  // This enables batch write flow control
+
+      mutations.apply(
+          String.format("Write data to table %s via BigtableIO", options.getBigtableTableId()),
+          write
+      );
+    }
+
+    p.run();
```

Comment:
Does it intend for run asynchronously? Please, append `waitUntilFinish()` call to the result of the `run()`.

Reply:
So far our example jobs all run async, is it a better practice to run sync? I'm happy to learn the reasoning of the practice.

Reply:
This is a code sample. It is a good practice to have the binary terminate _after_ the sampled behavior is complete.
If there is practical differences between implementing the batch flow control asynchronously and synchronously, consider creating multiple code samples that demonstrate the behaviors.

Reply:
Done.